#goal definition
string filePath
bool start
---
#result definition
bool success
---
#feedback
bool logging